R/default-internal-plot-configuration.R

```diff
@@ -0,0 +1,322 @@
+# class definition ---------------------------------
+
+#' @title Convenience class for plot configuration for OSP plots
+#'
+#' @description
+#'
+#' An intermediary R6 class that provides a single portal to prepare the
+#' `PlotConfiguration` object needed by `{tlf}` plotting functions. The specific
+#' type of `PlotConfiguration` objects can access all necessary objects formed
+#' from user inputs using this internal object.
+#'
+#' @field labels `tlf::LabelConfiguration` object defining properties of labels.
+#' @field legend `tlf::LegendConfiguration` object defining legend properties.
+#' @field xAxis `tlf::XAxisConfiguration` object defining `x`-axis properties.
+#' @field yAxis `tlf::YAxisConfiguration` object defining `y`-axis properties.
+#' @field background `tlf::BackgroundConfiguration` object defining the
+#'   configuration of background.
+#' @field lines `tlf::ThemeAestheticSelections` object or list defining how
+#'   lines are plotted.
+#' @field points `tlf::ThemeAestheticSelections` object or list defining how
+#'   points are plotted.
+#' @field ribbons `tlf::ThemeAestheticSelections` object or list defining
+#'   how ribbons are plotted.
+#' @field errorbars `tlf::ThemeAestheticSelections` object or list defining
+#'   how errorbars are plotted.
+#' @field export `tlf::SaveConfiguration` object defining saving properties.
+#'
+#' @keywords internal
+DefaultInternalPlotConfiguration <- R6::R6Class(
+  "DefaultInternalPlotConfiguration",
+  public = list(
+    labels = NULL,
+    legend = NULL,
+    xAxis = NULL,
+    yAxis = NULL,
+    background = NULL,
+    lines = NULL,
+    points = NULL,
+    ribbons = NULL,
+    errorbars = NULL,
+    export = NULL
+  )
+)
+
+
+# function to create instance ---------------------------------
+
+#' Create an instance of `defaultInternalPlotConfiguration` class
+#'
+#' @examples
+#'
+#' ospsuite:::.createDefaultInternalPlotConfiguration(DefaultPlotConfiguration$new())
+#'
+#' @keywords internal
+#' @noRd
+.createDefaultInternalPlotConfiguration <- function(defaultPlotConfiguration) {
+  # check if `{tlf}` is installed; will produce an error if the package is not installed
+  requireNamespace("tlf", quietly = TRUE)
+
+  validateIsOfType(defaultPlotConfiguration, "DefaultPlotConfiguration", nullAllowed = FALSE)
+
+  # labels object ---------------------------------------
+
+  labelTitle <- tlf::Label$new(
+    text = defaultPlotConfiguration$title,
+    font = NULL,
```

Comment:
why NULL for font?

Reply:
@pchelle can correct me if I am wrong here, but `font` parameter is redundant here. 

I am specifying `tlf::Label` object here which already specifies all aesthetic properties for the font

```r
tlf::Label$new(
  text = "",
  font = NULL,
  color = NULL,
  size = NULL,
  fontFace = NULL,
  fontFamily = NULL,
  angle = NULL,
  align = NULL
)
```

that `tlf::Font` object would do:

```r
tlf::Font$new(
  color = NULL,
  size = NULL,
  fontFamily = NULL,
  fontFace = NULL,
  angle = NULL,
  align = NULL
)
```

So, it is not necessary to specify argument for this parameter.